lib/ios/TopBarAppearancePresenter.m

```diff
@@ -0,0 +1,79 @@
+#import "TopBarAppearancePresenter.h"
+#import "RNNFontAttributesCreator.h"
+
+@interface TopBarAppearancePresenter ()
+
+@end
+
+
+@implementation TopBarAppearancePresenter {
+    UINavigationBarAppearance* _appearance;
+}
+@synthesize backgroundColor = _backgroundColor;
+@synthesize translucent = _translucent;
+
+- (instancetype)initWithNavigationController:(UINavigationController *)boundNavigationController {
+    self = [super initWithNavigationController:boundNavigationController];
+    _appearance = boundNavigationController.navigationBar.standardAppearance ?: [UINavigationBarAppearance new];
+    boundNavigationController.navigationBar.standardAppearance = _appearance;
+    return self;
+}
+
+- (void)updateBackgroundAppearance {
```

Comment:
Lets move this function below the setters as they call it. That way we can read the class top down